### PR TITLE
Improved tests: nsg rule, publicip ,subnet, route-table

### PR DESCRIPTION
**Pull request by EvgenyAgafonchikov**
Added tests of types
* Default values
* Negative tests
* Custom scenario tests

Fixed:
* NSG rule default destinationPortRange type

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@EvgenyAgafonchikov__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>You've already signed the contribution license agreement. Thanks!</span>
        <span>We will now validate the agreement and then real humans will evaluate your PR.</span>

TTYL, AZPRBOT;

File: lib/commands/arm/network/constants._js [hunk 1]
```diff
@@ -37,7 +37,7 @@ module.exports = {
     levelDef: 'Full',
     portMin: 0,
     portMax: 65535,
-    portDef: 80,
+    portDef: '80',
```
Comment (MikhailTryakhov) [inline lib/commands/arm/network/constants._js]: why do we use strings here?

Comment (EvgenyAgafonchikov) [inline lib/commands/arm/network/constants._js]: [link](https://github.com/Azure/azure-rest-api-specs/blob/master/arm-network/2016-09-01/swagger/network.json#L6763)

**Comment by amarzavery** [comment]
@EvgenyAgafonchikov @MikhailTryakhov - Resolve merge conflicts

**Comment by EvgenyAgafonchikov** [comment]
@amarzavery Conflicts are resolved

**Comment by amarzavery** [comment]
@EvgenyAgafonchikov - Sorry but the conflicts are back as I merged other PRs. Can you please resolve them again?

**Pull request merged**